astropy/nddata/tests/test_utils.py

```diff
@@ -78,6 +78,68 @@ def test_slices_overlap_wrong_mode():
     assert "Mode can be only" in str(e.value)
 
 
+def test_extract_array_even_shape_rounding():
+    """
+    Test overlap_slices (via extract_array) for rounding with an
+    even-shaped extraction.
+    """
+
+    data = np.arange(10)
+    shape = (2,)
+    positions = (1.49, 1.5, 1.501, 1.99, 2.0, 2.01, 2.49, 2.5, 2.501, 2.99,
+                 3.0, 3.01)
+    exp1 = (1, 2)
+    exp2 = (2, 3)
+    exp3 = (3, 4)
+    expected = [exp1, ] * 5 + [exp2, ] * 6 + [exp3, ]
```

Comment:
This set of inputs/expected outputs is not super easy to parse - I wonder whether it would be easier to just construct the zipped list manually e.g.

```python
positions_expected = [(1.49, (1, 2)),
                      (1.5), (1, 2)),
```

or something like that?

Reply:
Done!

Reply:
Done.